packages/aws-cdk/test/integ/cli/common.bash

```diff
@@ -13,7 +13,16 @@ fi
 
 
 if [[ "${STACK_NAME_PREFIX:-}" == "" ]]; then
-  if ${IS_CANARY:-false}; then
+  # Make the stack names unique based on the codebuild project name
+  # (if it exists). This prevents multiple codebuild projects stomping
+  # on each other's stacks and failing them.
+  #
+  # The get codebuild project name from the ID: PROJECT_NAME:1238a83
+  CODEBUILD_PROJECT=$(echo ${CODEBUILD_BUILD_ID:-} | cut -d: -f 1)
+
+  if [[ "${CODEBUILD_PROJECT:-}" != "" ]]; then
+    export STACK_NAME_PREFIX="${CODEBUILD_PROJECT}"
+  elif ${IS_CANARY:-false}; then
```

Comment:
What, canaries run as a codebuild project. This means that this is a dead condition, no? Maybe we can just omit it?

Reply:
I didn't want to make the assumption that this script is always going to be run in CodeBuild. The logic goes as follows:

* If CodeBuild, use project name.
* If not, use old naming scheme depending on whether we are a canary or not.

Reply:
okay cool

Reply:
Run-by-hand in canary mode is a thing we may occasionally have to do?